domain/schema/controller/sql/0021-cloudimagemetadata.sql

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE cloud_image_metadata (
+    created_at TIMESTAMP NOT NULL,
+    source TEXT NOT NULL,
+    stream TEXT NOT NULL,
+    region TEXT NOT NULL,
+    version TEXT NOT NULL,
+    arch TEXT NOT NULL,
```

Comment:
This should be a foreign key to the architecture table. We don't want any arches that we don't support.

https://github.com/juju/juju/blob/fb6d5cf729bebf5cb4728d09b7ce115c49320a84/domain/schema/model/sql/0007-platform.sql#L12-L25

Reply:
I change it according to other snippet in the code using the architecture_id

**question:** However, since we have an index on `architecture.name`, wouldn't be better to use directly the name ? (avoid a join whenever write or read table with architecture and more readable when doing a manual `select` on it)